### PR TITLE
376 gippy update

### LINKED ISSUE #376: migrate gips from gippy==0.3.* to gippy==1.0.0
(Note that effort was made in 2017 that was then largely abandoned; the work restarts in circa 2019; see below.)

### PR DESCRIPTION
**Pull request by ags-tolson**
Fixes #376.  Pretty much every single gippy call in gips is changing.  Most products are the same as before, but sometimes with changes to metadata that didn't seem incorrect.  A few are broken or otherwise unexpectedly different.  See #376 for details.

Most of the calls are just spelt different, eg `getProjection` is now `get_srs`; a few are method calls instead of constructors, such as `GeoImage.create_from(...)` instead of `GeoImage(the, version, with, four_args)`. 
 In a few places it made sense to refactor the code before conversion, and to add minor features to core code for convenience.

### CODE REVIEW

File: gips/data/weld/weld.py [hunk 1]
```diff
@@ -181,6 +196,7 @@ def process(self, *args, **kwargs):
 
             meta = self.meta_dict()
             meta['AVAILABLE_ASSETS'] = ''
+            meta['VERSION'] = '1.0'
```
Comment (ircwaves) [inline gips/data/weld/weld.py]: not required, but seems like this should use `self.version`

File: install.sh [hunk 2]
```diff
@@ -33,8 +33,8 @@ source .venv/bin/activate
```
Comment (ircwaves) [inline install.sh]: dependencies (a few lines back): drop `libboost-all-dev`

File: gips/tiles.py [hunk 3]
```diff
@@ -92,14 +94,13 @@ def mosaic(self, datadir, res=None, interpolation=0, crop=False,
...
-                        CookieCutter(
-                            images, self.spatial.site, fout, res[0], res[1],
-                            crop, interpolation, {}, alltouch,
-                        )
-                    else:
+                    images = [gippy.GeoImage(f) for f in filenames]
+                    if None in (res, self.spatial.site):
                         mosaic(images, fout, self.spatial.site)
+                    else:
+                        # "" == default to the projection used by the GeoFeature self.spatial.site
```
Comment (ircwaves) [inline gips/tiles.py]: good note

**Comment by ircwaves** [comment]
cannot merge tho.

File: gips/data/landsat/landsat.py [hunk 4]
```diff
@@ -864,18 +865,15 @@ def process(self, products=None, overwrite=False, **kwargs):
                     cfmask[cfmask == 2] = 0
 
                     verbose_out("writing " + fname, 2)
-                    imgout = gippy.GeoImage(fname, img, gippy.GDT_Byte, 1)
-                    imgout.SetBandName('Land mask', 1)
-                    imgout[0].Write(cfmask)
+                    imgout = gippy.GeoImage(img, fname, 1, 'byte')
```
Comment (ags-tolson) [inline gips/data/landsat/landsat.py]: Oops; I think I forgot the `create_from`; this needs to be `imgout = gippy.GeoImage.create_from(img, fname, 1, 'byte')`

File: gips/data/landsat/landsat.py [hunk 5]
```diff
@@ -971,6 +968,7 @@ def get_bit(np_array, i):
                             """Return an array with the ith bit extracted from each cell."""
                             return (np_array >> i) & 0b1
 
+                        # here down is dev version; needs update for gippy 1.0
```
[the remark this answers is not in the document]
Comment (ags-tolson) [reply gips/data/landsat/landsat.py]: I think this is a stale comment; I'll remove it.

File: gips/data/landsat/landsat.py [hunk 6]
```diff
@@ -935,32 +933,31 @@ def process(self, products=None, overwrite=False, **kwargs):
                         continuable=True):
                     fname = self.temp_product_filename(sensor, key)
                     if val[0] == 'acca':
+                        raise NotImplementedError("ACCA is currently not functional with gippy 1.0")
```
[the remark this answers is not in the document]
Comment (ags-tolson) [reply gips/data/landsat/landsat.py]: I re-ran the acca product just now and got an exception; it doesn't seem to be related to a missing mask; it looks more like an array index error kinda thing.  Should I change the exception message?  Error message:
```
RuntimeError: error writing /home/tolson/src/also-gips/data-root/landsat/stage/procEdpE56/012030_2017181_LC8_acca.tif, band 2: Access window out of range in RasterIO().  Requested
(-53,-53) of size 7987x2234 on raster of 7881x7991.
```

More context:
```
Warm Cloud stats(min,max,mean,sd,skew,count): this = 0x7ffd36e3fa80, size = (6,1,1,1) [24 b], data = (float*)0x8e2f360..0x8e2f377 (non-shared) = [ -7.276953220367432 7.919604301452637 0.5986603498458862 4.18020486831665 -0.05579585582017899 3624851 ], min = -7.27695, max = 3.62485e+06, mean = 604143, std = 1.47984e+06, coords_min = (0,0,0,0), coords_max = (5,0,0,0).
distance = 1956.17
dx       = -47
dy       = -45
smearlen = 65
dilate = 10
xstep  = -1
ystep  = -1
012030_2017181_LC8_acca[pass1]: read (0,0)-(7881,2128) in 0.02587 seconds
LC08_L1TP_012030_20170630_20170715_01_T1_B10[LWIR]: read (0,0)-(7881,2181) in 1.394 seconds
LC08_L1TP_012030_20170630_20170715_01_T1_B10[LWIR]: read (0,0)-(7881,2181) in 1.198 seconds
012030_2017181_LC8_acca[ambclouds]: read (0,0)-(7881,2128) in 2.689 seconds
012030_2017181_LC8_acca[cloudmask]: Writing (1x + 0)
Error creating product acca for LC08_L1TP_012030_20170630_20170715_01_T1.tar:
Traceback (most recent call last):
  File "/home/tolson/src/also-gips/gips/utils.py", line 494, in cli_error_handler
    yield
  File "/home/tolson/src/also-gips/gips/data/landsat/landsat.py", line 948, in process
    s_azim, erosion, dilation, cloudheight)
  File "/home/tolson/src/also-gips/.g1.0venv/local/lib/python2.7/site-packages/gippy/algorithms.py", line 89, in acca
    return _algorithms.acca(*args, **kwargs)
RuntimeError: error writing /home/tolson/src/also-gips/data-root/landsat/stage/procEdpE56/012030_2017181_LC8_acca.tif, band 2: Access window out of range in RasterIO().  Requested
(-53,-53) of size 7987x2234 on raster of 7881x7991.
```

File: install.sh (hunk 2 above)
Comment (ags-tolson) [reply install.sh]: see 3d30b2b

**Comment by ags-tolson** [comment]
Don't merge yet; there's a problem with `t_process_acolite`; running the landsat system test for it now.

File: gips/data/landsat/landsat.py [hunk 7]
```diff
@@ -1265,10 +1267,10 @@ def _readqa(self):
             qadatafile = self.assets[asset].extract([self.metadata['qafilename']])
         else:
             # Use tar.gz directly using GDAL's virtual filesystem
-            qadatafile = os.path.join('/vsitar/' + self.assets[asset].filename, self.metadata['qafilename'])
-        qaimg = gippy.GeoImage(qadatafile)
-        return qaimg
-
+            qadatafile = os.path.join('/vsitar/' + self.assets[asset].filename,
+                                      self.metadata['qafilename'])
+        npqa = gippy.GeoImage(qadatafile).read().astype('uint16')
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: Looks like `read_raw` will handle this w/o the conversion.

Comment (ags-tolson) [reply gips/data/landsat/landsat.py]: Unfortunately it looks like that's not exposed to python; `gippy.GeoImage(qadatafile).read_raw` right there caused an AttributeError when I tried it just now.

Comment (ircwaves) [reply gips/data/landsat/landsat.py]: it is  a method of `GeoRaster`, so 
```
>>> import gippy as gp
>>> i = gp.GeoImage.create()
>>> a = i[0].read_raw()
>>> a.dtype
dtype('uint8')
>>> i.type().string()
'uint8'
```

Comment (ags-tolson) [reply gips/data/landsat/landsat.py]: So technically it's both; I didn't think to go look in GeoRaster when the GeoImage call failed.  I'll try again after standup.

**Pull request closed without merging**